google-mysql.yml

```diff
@@ -168,6 +168,11 @@ provision:
     default: ""
     constraints:
       pattern: ^[a-z]?$
+  - field_name: allow_insecure_connections
+    type: boolean
+    details: Allow insecure connections between the application and this service instance
+    default: false
+    prohibit_update: true
```

Comment:
Why? I can imagine a pattern when the first step is to get the app working without TLS, then the app is updated to use TLS, and then the admin wants to enforce secure connections. It would be useful to be able to upgrade the security - even if it's only for new bindings. But there may be other arguments, and there may be technical limitations.

Reply:
There are no technical limitations, but there is no way to ensure that there are no insecure bindings, as it's possible to update the property both ways anytime, so there can be no guarantees.

Reply:
We had this discussion with @jhvhs the other day. He brought up an interesting perspective. If you create the instance with `allow_insecure_connections=true` then create bindings and then update to `allow_insecure_connections=false`, then it might be inferred that the old bindings would not work anymore, or that they would also be forced to use secure connections. However that is not the case, old bindings will still permit insecure connections. Which can be confusing and a security concern if not deleted.